src/Inventory.zig

```diff
@@ -359,6 +373,16 @@ pub const Sync = struct { // MARK: Sync
 			return null;
 		}
 
+		pub fn getInventoryFromUserAndSource(user: *main.server.User, source: SourceType) ?Inventory {
+			main.utils.assertLocked(&mutex);
+			for(inventories.items) |inv| {
+				if (std.mem.containsAtLeast(*main.server.User, inv.users.items, 1, &[_]*main.server.User{user}) and inv.source == source) {
```

Comment:
Maybe we should store the user in the SourceType for the player hand and inventory?

The current solution would cause trouble if we ever want players to be able to open other player's inventorys. (like e.g. the admin checking your inventory)